src/test/javascript/spec/component/programming-assessment/programming-assessment-repo-export-dialog.component.spec.ts

```diff
@@ -0,0 +1,128 @@
+import { Course } from 'app/entities/course.model';
+import { ExerciseService } from 'app/exercises/shared/exercise/exercise.service';
+import { ProgrammingAssessmentRepoExportService } from 'app/exercises/programming/assess/repo-export/programming-assessment-repo-export.service';
+import { ArtemisProgrammingAssessmentModule } from 'app/exercises/programming/assess/programming-assessment.module';
+import { Exercise } from 'app/entities/exercise.model';
+import { MockSyncStorage } from '../../helpers/mocks/service/mock-sync-storage.service';
+
+chai.use(sinonChai);
+const expect = chai.expect;
+
+const createBlobHttpResponse = () => {
+    const blob = new Blob([JSON.stringify({ property: 'blob' })], { type: 'application/json' });
+    const headers = new HttpHeaders().set('filename', 'blobfile');
+    return new HttpResponse({ body: blob, headers });
+};
+
+describe('ProgrammingAssessmentRepoExportDialogComponent', () => {
+    let comp: ProgrammingAssessmentRepoExportDialogComponent;
+    let fixture: ComponentFixture<ProgrammingAssessmentRepoExportDialogComponent>;
+    let exerciseService: ExerciseService;
+    let repoExportService: ProgrammingAssessmentRepoExportService;
+
+    // stubs
+    let findExerciseId: SinonStub;
+    global.URL.createObjectURL = jest.fn(() => 'http://some.test.com');
+    global.URL.revokeObjectURL = jest.fn(() => '');
+
+    const exerciseId = 42;
+    const participationIdList = [1];
+    const singleParticipantMode = false;
+    const programmingExercise = new ProgrammingExercise(new Course(), undefined);
+    programmingExercise.id = exerciseId;
+    programmingExercise.releaseDate = moment();
+    programmingExercise.dueDate = moment().add(7, 'days');
+
+    beforeEach(async () => {
+        return TestBed.configureTestingModule({
+            imports: [TranslateModule.forRoot(), ArtemisTestModule, ArtemisProgrammingAssessmentModule],
+            providers: [
+                ExerciseService,
+                ProgrammingAssessmentRepoExportService,
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+            ],
+        })
+            .overrideModule(ArtemisTestModule, { set: { declarations: [], exports: [] } })
+            .compileComponents()
+            .then(() => {
+                // Ignore console errors
+                console.error = () => false;
+                fixture = TestBed.createComponent(ProgrammingAssessmentRepoExportDialogComponent);
+                comp = fixture.componentInstance;
+                exerciseService = fixture.debugElement.injector.get(ExerciseService);
+                repoExportService = fixture.debugElement.injector.get(ProgrammingAssessmentRepoExportService);
+
+                // stubs
+                findExerciseId = stub(exerciseService, 'find').returns(of({ body: programmingExercise } as HttpResponse<Exercise>));
```

Comment:
The `findExerciseId` variable doesn't seem to be used. (If you need the stub statement I'd just remove the variable assignment.)